Blog post of Android Tracker 1.3.0 (#1302)
snowplow/snowplowanalytics.com: _posts/2019-09-09-snowplow-android-tracker-1.3.0-released/index.md

---
layout: post
title: "Snowplow Android Tracker 1.3.0 released"
title-short: Snowplow Android Tracker 1.3.0
tags: [snowplow, android, java, mobile]
author: Oguzhan
category: Releases
permalink: /blog/2019/09/09/snowplow-android-tracker-1.3.0-released/
discourse: true
---

We are pleased to announce a new release of the [Snowplow Android Tracker][repo].

> [Version 1.3.0][release-notes] introduces integration of Snowplow Micro and compliance with targeted ads preference of Android users.

Reviewer: ```suggestion
[Version 1.3.0][release-notes] introduces a Snowplow Micro powered test suite and compliance with targeted ads preference of Android users.
```

> With this release, Snowplow users can set up their tracking more reliably so your data team can gain access to valuable insights about your application.

Reviewer: ```suggestion
With this release, we're building assurance that engineers can set up their tracking more reliably and your data team can have uninterrupted access to valuable insights about your application.
```

Read on below the fold for:

1. [Integration of Snowplow Micro](#micro-integration)
2. [Compliance with targeted ads preference](#ad-tracking)
3. [Updates](#updates)
4. [Documentation](#docs)
5. [Getting help](#help)

<!--more-->

<h2 id="micro-integration">1. Integration of Snowplow Micro</h2>

> Our recent tracker releases have shown that we need to increase our confidence in new releases. The latest example was Android Tracker 1.2.0 where a small implementation mistake caused the duplication of natively supported contexts in case they are enabled, a mistake that could be prevented if we had a more comprehensive test suit, hence the integration of Snowplow Micro was more than necessary.

Reviewer: ```suggestion
Our recent tracker releases have shown that we need to increase confidence in new releases. The latest example was Android Tracker 1.2.0 where a small implementation mistake caused the duplication of natively supported contexts. A mistake that could be prevented if we had a more comprehensive test suite, which we can do with the newly arrived Snowplow Micro project.
```

> In essence, Snowplow Micro is a squeezed Snowplow pipeline, built to be used in automated test suite of any tracking setup to increase the confidence. On our side, Micro will allow us to release reliable trackers. We started writing integration tests on our demo app, to be extended for each new feature we'll introduce in our tracker. Please read [the blog post][micro-blog-post] to learn more about Snowplow Micro. 

Reviewer: ```suggestion
In essence, Snowplow Micro is our smallest Snowplow pipeline, built specifically to be used in the automated test suite of any tracking setup to increase confidence in implementation. While users may use Micro to test their tracking strategies, at Snowplow we use Micro to test that we are releasing reliable trackers. We started writing integration tests on our demo app, to be extended for each new feature we'll introduce in our tracker. Please read [the blog post][micro-blog-post] to learn more about Snowplow Micro. 
```

<h2 id="targeted-ads-preference">2. Compliance with targeted ads preference</h2>

> On late 2013, Google [introduced][google-play-services-4] a new approach on advertising, including a new Advertising ID, to give users greater control such as resetting their device's Ads ID or opt out of interest-based, a.k.a. targeted, ads.

Reviewer: ```suggestion
In late 2013, Google [introduced][google-play-services-4] a new approach for advertising, including a new Advertising ID, to give users greater control such as resetting their device's Ads ID or opt out of interest-based (a.k.a. targeted) ads.
```

> On our side, we've been offering the ability to collect Ads ID since mid-2015 with a caveat. Our tracker would fetch Ads ID regardless of user preference, meaning that Snowplow Android Tracker users have always been able to collect their users' Ads ID. As of this release, our tracker will respect Android users' preference and not fetch Ads ID if they opt out of targeted ads.

Reviewer: ```suggestion
As of this release, our tracker will respect Android users' preference and not fetch Ads ID if they opt out of targeted ads.
```
